### PR TITLE
 Port tokenizer and preprocessing tests to run_preprocessing_layer_test

### PR DESCRIPTION
**Pull request by Ahmed0830**
This PR addresses #1313.

### Tokenizer tests

Ports these files to use `TestCase.run_preprocessing_layer_test` as their first test, removing `tf.data` map/batch tests already covered by the shared helper, and removing `test_config`/`test_load_model_with_config` methods that duplicated the `get_config`/`from_config` round-trip check `run_preprocessing_layer_test`'s `run_serialization_test` already performs:

- `byte_tokenizer_test.py`
- `unicode_codepoint_tokenizer_test.py`
- `word_piece_tokenizer_test.py`
- `byte_pair_tokenizer_test.py`
- `sentence_piece_tokenizer_test.py`

In `byte_pair_tokenizer_test.py` and `sentence_piece_tokenizer_test.py`, also removes test overrides in `BytePairTokenizerTFTest`/`SentencePieceTokenizerTFTest` that were duplicates of the base tests aside from a hardcoded `_allow_python_workflow=False`. The base tests now read `self._allow_python_workflow` instead, so they already run under both workflows via inheritance.

### Preprocessing layer tests

Also ports six keras_hub/src/layers/preprocessing/*_test.py files:

- `image_converter_test.py`: replaces `test_resize_dataset` and removes the redundant `test_config`.
- `masked_lm_mask_generator_test.py`: replaces `test_mask_dense` and `test_with_tf_data` with one deterministic case (mask_selection_rate=1, mask_token_rate=1, random_token_rate=0), the layer holds a stateful tf.random.Generator, so non-deterministic settings can't be reused across the helper's repeated calls and removes the redundant `test_config`.
- `start_end_packer_test.py`: replaces `test_batch`, now parameterized over `_allow_python_workflow` like the rest of the file.
- `multi_segment_packer_test.py`: adds `test_layer_basics`, removes the redundant `test_config`. call() takes a single inputs argument (the tuple of segments), so input_data needs an extra 1-tuple wrapper to match the helper's layer(*input_data) convention.
- `random_deletion_test.py`: adds `test_layer_basics` (rate=1.0, deletes every token, deterministic regardless of RNG state), removes the redundant `test_get_config_and_from_config`, and trims `test_augment_first_batch_second`/`test_batch_first_augment_second` down to just their `skip_fn`/`skip_py_fn` cases, since the default no-skip case is now covered generically by `test_layer_basics` but `skip_fn`/`skip_py_fn` are distinct code paths that need their own verification under tf.data.Dataset.map.
- `random_swap_test.py`: same treatment, using rate=0.0 (never swaps) since, unlike deletion, swap-partner selection has no rate that makes the result order-independent, 0.0 is the only fully deterministic option.

### Note
`test_layer_basics` for `RandomDeletion`/`RandomSwap` uses int tokens rather than these layers' more typical string/word-level usage. `run_preprocessing_layer_test`'s internal direct-call-vs-tf.data-mapped consistency check uses assertAllClose, which calls np.isclose internally; np.isclose tries to promote its inputs against a float tolerance and raises DTypePromotionError for string dtypes, even when comparing two identical strings, so the helper's consistency check is currently broken for any string-output layer, independent of test data. This looks like a latent bug in `test_case.py` but would like to understand if this is intentional.

### CODE REVIEW

File: keras_hub/src/tokenizers/byte_pair_tokenizer_test.py [hunk 1]
```diff
@@ -18,10 +18,29 @@
...
+            },
+            input_data=[
+                "I am just a test string",
+                "I am also a test string",
+            ],
+            expected_output=[
+                [100, 524, 95, 10, 1296, 6755],
+                [100, 524, 67, 10, 1296, 6755],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/byte_pair_tokenizer_test.py:44]: I don't think this is needed, as the run_preprocessing_layer_test checks if expected_detokenize_output is None, if it is, then expected_detokenize_output is nothing but the input_data. So the above suggested change wouldn't change the outcome of this test

File: keras_hub/src/tokenizers/byte_tokenizer_test.py [hunk 2]
```diff
@@ -5,18 +5,34 @@
...
+            cls=ByteTokenizer,
+            init_kwargs={},
+            input_data=["hello", "fun", "▀▁▂▃", "haha"],
+            expected_output=[
+                [104, 101, 108, 108, 111],
+                [102, 117, 110],
+                [226, 150, 128, 226, 150, 129, 226, 150, 130, 226, 150, 131],
+                [104, 97, 104, 97],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/byte_tokenizer_test.py:19]: same here

File: keras_hub/src/tokenizers/byte_tokenizer_test.py [hunk 3]
```diff
@@ -5,18 +5,34 @@
...
+            cls=ByteTokenizer,
+            init_kwargs={"sequence_length": 12},
+            input_data=["hello", "fun", "▀▁▂▃", "haha"],
+            expected_output=[
+                [104, 101, 108, 108, 111, 0, 0, 0, 0, 0, 0, 0],
+                [102, 117, 110, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [226, 150, 128, 226, 150, 129, 226, 150, 130, 226, 150, 131],
+                [104, 97, 104, 97, 0, 0, 0, 0, 0, 0, 0, 0],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/byte_tokenizer_test.py:35]: same here

File: keras_hub/src/tokenizers/sentence_piece_tokenizer_test.py [hunk 4]
```diff
@@ -12,11 +11,33 @@
...
+                "the",
+                "quick brown fox.",
+            ],
+            expected_output=[
+                [6, 5, 3, 4],
+                [6, 5],
+                [6],
+                [5, 3, 4],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/sentence_piece_tokenizer_test.py:42]: same here

File: keras_hub/src/tokenizers/unicode_codepoint_tokenizer_test.py [hunk 5]
```diff
@@ -7,18 +7,35 @@
...
+            init_kwargs={},
+            input_data=["ninja", "samurai", "▀▁▂▃", "keras", "tensorflow"],
+            expected_output=[
+                [110, 105, 110, 106, 97],
+                [115, 97, 109, 117, 114, 97, 105],
+                [9600, 9601, 9602, 9603],
+                [107, 101, 114, 97, 115],
+                [116, 101, 110, 115, 111, 114, 102, 108, 111, 119],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/unicode_codepoint_tokenizer_test.py:22]: same here

File: keras_hub/src/tokenizers/unicode_codepoint_tokenizer_test.py [hunk 6]
```diff
@@ -7,18 +7,35 @@
...
+            init_kwargs={"sequence_length": 10},
+            input_data=["ninja", "samurai", "▀▁▂▃", "keras", "tensorflow"],
+            expected_output=[
+                [110, 105, 110, 106, 97, 0, 0, 0, 0, 0],
+                [115, 97, 109, 117, 114, 97, 105, 0, 0, 0],
+                [9600, 9601, 9602, 9603, 0, 0, 0, 0, 0, 0],
+                [107, 101, 114, 97, 115, 0, 0, 0, 0, 0],
+                [116, 101, 110, 115, 111, 114, 102, 108, 111, 119],
+            ],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/tokenizers/unicode_codepoint_tokenizer_test.py:38]: same here

File: keras_hub/src/layers/preprocessing/start_end_packer_test.py [hunk 7]
```diff
@@ -6,6 +6,24 @@
...
+            init_kwargs={
+                "sequence_length": 7,
+                "start_value": 1,
+                "end_value": 2,
+                "pad_value": 3,
+                "_allow_python_workflow": allow_python_workflow,
+            },
+            input_data=tf.ragged.constant([[5, 6, 7], [8, 9, 10, 11]]),
+            expected_output=[[1, 5, 6, 7, 2, 3, 3], [1, 8, 9, 10, 11, 2, 3]],
+        )
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/layers/preprocessing/start_end_packer_test.py:25]: A plain nested list won't work here, run_preprocessing_layer_test calls tf.data.Dataset.from_tensor_slices(input_data), and since the rows have different lengths, a raw Python list raises ValueError: Can't convert non-rectangular Python sequence to Tensor. tf.ragged.constant is needed for that, so I'll leave it as is.

**Review by laxmareddyp — CHANGES_REQUESTED** [review]
Thanks for cleanup..!

the deleted test_config / test_load_model_with_config tests in unicode_codepoint_tokenizer_test.py and word_piece_tokenizer_test.py were testing from_config round-trips with non-default params (e.g., lowercase=False, normalization_form="NFC", vocabulary_size=100, special_tokens=[...], suffix_indicator="@@"). The new test_tokenizer_basics only passes default init_kwargs, so run_serialization_test will only round-trip the default config. 

Can we keep a test_tokenizer_basics variant with non-default kwargs for these two tokenizers to preserve that coverage?

Ex:

```
def test_tokenizer_basics_with_non_default_config(self):
    self.run_preprocessing_layer_test(
        cls=UnicodeCodepointTokenizer,
        init_kwargs={
            "lowercase": False,
            "sequence_length": 8,
            "normalization_form": "NFC",
            "errors": "ignore",
            "replacement_char": 0,
            "vocabulary_size": 100,
        },
        input_data=["ninja", "samurai"],
        expected_output=[[78, 73, 78, 74, 65, 0, 0, 0], [83, 65, 77, 0, 0, 0, 0, 0]],
    )
```
Please do resolve the conflicts. Thanks

**Comment by Ahmed0830** [comment]
> Thanks for cleanup..!
> 
> the deleted test_config / test_load_model_with_config tests in unicode_codepoint_tokenizer_test.py and word_piece_tokenizer_test.py were testing from_config round-trips with non-default params (e.g., lowercase=False, normalization_form="NFC", vocabulary_size=100, special_tokens=[...], suffix_indicator="@@"). The new test_tokenizer_basics only passes default init_kwargs, so run_serialization_test will only round-trip the default config.
> 
> Can we keep a test_tokenizer_basics variant with non-default kwargs for these two tokenizers to preserve that coverage?
> 
> Ex:
> 
> ```
> def test_tokenizer_basics_with_non_default_config(self):
>     self.run_preprocessing_layer_test(
>         cls=UnicodeCodepointTokenizer,
>         init_kwargs={
>             "lowercase": False,
>             "sequence_length": 8,
>             "normalization_form": "NFC",
>             "errors": "ignore",
>             "replacement_char": 0,
>             "vocabulary_size": 100,
>         },
>         input_data=["ninja", "samurai"],
>         expected_output=[[78, 73, 78, 74, 65, 0, 0, 0], [83, 65, 77, 0, 0, 0, 0, 0]],
>     )
> ```
> 
> Please do resolve the conflicts. Thanks

Thanks for the review @laxmareddyp. I have added the tests for non default kwargs and have resolved the conflicts. Please let me know if there's any other changes required.

**Review by laxmareddyp — APPROVED** [review]
Thanks..! LGTM

**Comment by divyashreepathihalli** [comment]
@jules can you verify the changes in this PR?
